SpatialGDK/Source/SpatialGDKServices/Private/SSpatialOutputLog.cpp

```diff
@@ -125,6 +125,15 @@ void SSpatialOutputLog::OnLogDirectoryChanged(const TArray<FFileChangeData>& Fil
 	}
 }
 
+void SSpatialOutputLog::OnClearLog()
+{
+	// SOutputLog will clear the messages and the SelectedLogCategories.
+	SOutputLog::OnClearLog();
+
+	// Also clear the AvailableLogCategories as we generate many worker categories which are hard to parse.
+	Filter.AvailableLogCategories.Reset(Filter.AvailableLogCategories.GetAllocatedSize());
```

Comment:
```suggestion
	Filter.AvailableLogCategories.Reset();
```

Reply:
Based this on an optimisation in the Engine for the SelectedLogCategories.
```
void FLogFilter::ClearSelectedLogCategories()
{
	// No need to churn memory each time the selected categories are cleared
	SelectedLogCategories.Reset(SelectedLogCategories.GetAllocatedSize());
}
```

Reply:
That's not an optimization, it's a bug. This will increase the array's allocated buffer every time. It gets the size in bytes and tells the array to fit that many elements every time.
Here's the output I got from a simple test:
```
TArray<int> test;
test.Add(5);
for (int i = 0; i < 20; i++)
{
	UE_LOG(LogTemp, Warning, TEXT("Array allocated size: %d"), test.GetAllocatedSize());
	test.Reset(test.GetAllocatedSize());
}
```
```
[2019.10.07-20.01.26:975][477]LogTemp: Warning: Array allocated size: 16
[2019.10.07-20.01.26:975][477]LogTemp: Warning: Array allocated size: 64
[2019.10.07-20.01.26:976][477]LogTemp: Warning: Array allocated size: 256
[2019.10.07-20.01.26:976][477]LogTemp: Warning: Array allocated size: 1024
[2019.10.07-20.01.26:976][477]LogTemp: Warning: Array allocated size: 4096
[2019.10.07-20.01.26:976][477]LogTemp: Warning: Array allocated size: 16384
[2019.10.07-20.01.26:977][477]LogTemp: Warning: Array allocated size: 65536
[2019.10.07-20.01.26:977][477]LogTemp: Warning: Array allocated size: 262144
[2019.10.07-20.01.26:977][477]LogTemp: Warning: Array allocated size: 1048576
[2019.10.07-20.01.26:977][477]LogTemp: Warning: Array allocated size: 4194304
[2019.10.07-20.01.26:979][477]LogTemp: Warning: Array allocated size: 16777216
[2019.10.07-20.01.26:986][477]LogTemp: Warning: Array allocated size: 67108864
[2019.10.07-20.01.27:015][477]LogTemp: Warning: Array allocated size: 268435456
[2019.10.07-20.01.27:132][477]LogTemp: Warning: Array allocated size: 1073741824
[2019.10.07-20.01.27:610][477]LogTemp: Warning: Array allocated size: 0
[2019.10.07-20.01.27:610][477]LogTemp: Warning: Array allocated size: 0
[2019.10.07-20.01.27:610][477]LogTemp: Warning: Array allocated size: 0
[2019.10.07-20.01.27:610][477]LogTemp: Warning: Array allocated size: 0
[2019.10.07-20.01.27:611][477]LogTemp: Warning: Array allocated size: 0
[2019.10.07-20.01.27:611][477]LogTemp: Warning: Array allocated size: 0
```

Reply:
:o